libs/galley-types/src/Galley/Types/Teams/Intra.hs

```diff
@@ -59,3 +62,8 @@ instance FromJSON TeamStatusUpdate where
 
 instance ToJSON TeamStatusUpdate where
     toJSON s = object ["status" .= tuStatus s]
+
+newtype TeamName = TeamName
+    { tnName :: Text }
+
+deriveJSON toFieldName ''TeamName
```

Comment:
I prefer the Generic version for deriving the instances. I have no evidence, but something tells me it is less "Template Haskell-heavy". The field modifier can still be used like this:

```Haskell
instance ToJSON TeamName where
  toJSON = genericToJSON defaultOptions {
             fieldLabelModifier = toFieldName }
```
or something like that. I found that example here:
https://artyom.me/aeson

Reply:
We agreed, offline, to go with the TH version and will revisit this topic later 💃